tests/conftest.py

```diff
@@ -54,14 +64,30 @@ def execute(command, user=None, **kwargs):
             command = f"exec -T --user={user} aiidalab {command}"
         else:
             command = f"exec -T aiidalab {command}"
-        return docker_compose.execute(command, **kwargs)
+        out = docker_compose.execute(command, **kwargs)
+        return out.decode()
 
     return execute
 
 
 @pytest.fixture
 def nb_user(aiidalab_exec):
-    return aiidalab_exec("bash -c 'echo \"${NB_USER}\"'").decode().strip()
+    return aiidalab_exec("bash -c 'echo \"${NB_USER}\"'").strip()
+
+
+@pytest.fixture
+def pip_install(aiidalab_exec, nb_user):
+    """Temporarily install package via pip"""
+    package = None
+
+    def _pip_install(pkg, **args):
+        nonlocal package
+        package = pkg
+        return aiidalab_exec(f"pip install {pkg}", **args)
+
+    yield _pip_install
+    if package:
+        aiidalab_exec(f"pip uninstall --yes {package}")
```

Comment:
I remember I mentioned somewhere there is an issue that the uninstallation will only remove the package but not the dependencies. Is this fixture solve that?

Reply:
Good point, it doesn't. I am not sure if this is a problem in practice: Here I only use it for a package that I specifically chosen to have zero dependencies. But it would be good to at least document this.

I think it was a slightly bigger issue when running the integration tests and installing actuall AiiDAlab apps.

Reply:
> I think it was a slightly bigger issue when running the integration tests and installing actuall AiiDAlab apps.

You mean we should not test the QE-app or AWB in the full-stack image? That what I proposed but you said you were to comfortable with a image that not working with QeApp. 
Can we make it a nightly test for QeApp and AWB that runs periodically?

Reply:
> But it would be good to at least document this.

Yep, I'll add a comment on it.

Reply:
>  Can we make it a nightly test for QeApp and AWB that runs periodically?

I don't think that is very developer friendly, I'd much rather now about issues rightaway. 

I will note however that originally these app-installation tests were completely separate from the others and marked as "integration" tests, they specifically were not supposed to block merging. We should go back to that, I'll do that as part of #439, or as a separate PR after #442 is merged, which is a pre-requisite for this work.